standardtracer/tracer.go

```diff
@@ -2,16 +2,67 @@ package standardtracer
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
 
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-// New creates and returns a standard Tracer which defers to `recorder` after
-// RawSpans have been assembled.
-func New(recorder SpanRecorder) opentracing.Tracer {
+type sampleFunc func(int64) bool
+
+// Options allows creating a customized Tracer via NewWithOptions. The object
+// is thread safe and can be updated freely to influence the behavior of the
+// Tracer.
+type Options struct {
+	shouldSample       atomic.Value // contains a sampleFunc
+	trimUnsampledSpans int32        // updated atomically
+	recorder           atomic.Value // contains a SpanRecorder
```

Comment:
Don't worry, I appreciate the feedback and I'm really mostly testing the water here. Obviously I want a tracer that we can actually use at `cockroachdb`. We had some forking experiments when we implemented logging, and I value getting "grounded" in something that doesn't spiral off the common code base into an abyss of weirdness.

Stepping back, I think I can probably hide the dynamic options inside of the recorder and sampleFunc - I'll set `trimUnsampledSpans` unconditionally anyway. Getting rid of the atomic updates definitely doesn't make me feel bad.

I kept the `sampleFunc` simple because well - there simply isn't much information and I'm decidedly doing upfront sampling here since I don't want to pay the price for storing information that a smart sampler would later need. It's a start.

Regarding the forking, I thought that was going to happen anyway. As far as I'm concerned, now's as good as anytime. Maybe after the currently open PRs close.